cosipy/response/ExtendedSourceResponse.py

```diff
@@ -0,0 +1,125 @@
+from histpy import Histogram, Axes, Axis
+import numpy as np
+import astropy.units as u
+
+from .functions import get_integrated_extended_model
+
+class ExtendedSourceResponse(Histogram):
+    """
+    A class to represent and manipulate extended source response data.
+
+    This class provides methods to load data from HDF5 files, access contents,
+    units, and axes information, and calculate expectations based on sky models.
+
+    Methods
+    -------
+    get_expectation(allsky_image_model)
+        Calculate expectation based on an all-sky image model.
+    get_expectation_from_astromodel(source)
+        Calculate expectation from an astronomical model source.
+
+    Notes
+    -----
+    Currently, the axes of the response must be ['NuLambda', 'Ei', 'Em', 'Phi', 'PsiChi'].
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize an ExtendedSourceResponse object.
+        """
+        super().__init__(*args, **kwargs)
+        
+        if not np.all(self.axes.labels == ['NuLambda', 'Ei', 'Em', 'Phi', 'PsiChi']):
+            # 'NuLambda' should be 'lb' if it is in the gal. coordinates?
+            raise ValueError(f"The input axes {self.axes.labels} is not supported by ExtendedSourceResponse class.")
+
+    @classmethod
+    def open(cls, filename, name='hist'):
+        """
+        Load data from an HDF5 file.
+
+        Parameters
+        ----------
+        filename : str
+            The path to the HDF5 file.
+        name : str, optional
+            The name of the histogram group in the HDF5 file (default is 'hist').
+
+        Returns
+        -------
+        ExtendedSourceResponse
+            A new instance of ExtendedSourceResponse with loaded data.
+
+        Raises
+        ------
+        ValueError
+            If the shape of the contents does not match the axes.
+        """
+        hist = super().open(filename, name)
+
+        axes = hist.axes
+        contents = hist[:]
+        sumw2 = hist.sumw2 
+        unit = hist.unit
+        track_overflow = False
+        
+        new = cls(axes, contents = contents,
+                        sumw2 = sumw2,
+                        unit = unit,
+                        track_overflow = track_overflow)
+        
+        del hist
```

Comment:
I've changed the code corresponding to the first comment.

Reply:
> Why do you reconstruct the Histogram instead of using the opened one?

This is because the opened one is an instance of histpy.Histogram, not ExtendedSourceResponse. So, I needed to instantiate ExtendedSourceResponse at a new line. @israelmcmc Is there any nice way to directly open a hdf5 file as an instance of ExtendedSourceResponse?

Reply:
@hiyoneda I think that you can do `cls.open(filename, name)` instead of `super().open(filename, name)`. The output would be an `ExtendedSourceResponse` object.

Reply:
Nevermind. In principle this should work, but there's an issue with histpy that prevents it at the moment. I think this can be merged as is, and we can come back to it one I fix the issue in histpy

Reply:
> I've changed the code corresponding to the first comment.

Thanks @hirokiyoneda !

Reply:
> Nevermind. In principle this should work, but there's an issue with histpy that prevents it at the moment. I think this can be merged as is, and we can come back to it one I fix the issue in histpy

Sounds good!

Reply:
This is the corresponding issue, for the record https://gitlab.com/burstcube/histpy/-/issues/23